patches/tModLoader/Terraria/Localization/LanguageManager.tML.cs

```diff
@@ -148,4 +150,65 @@ private string ComputeBoundTextValue(TextBinding binding)
 		return string.Format(value, binding.args);
 	}
 	#endregion
+
+	private const int defaultWatcherCooldown = 15;
+	private FileSystemWatcher localizationFileWatcher;
+	internal string watchedMod;
+	internal string watchedFile;
+	internal int watcherCooldown;
+	private void SetupFileWatchers()
+	{
+		if (localizationFileWatcher != null)
+			return;
+		try {
+			localizationFileWatcher = new FileSystemWatcher();
+			localizationFileWatcher.Path = ModLoader.Core.ModCompile.ModSourcePath;
+			localizationFileWatcher.NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.FileName;
+			localizationFileWatcher.Filter = "*.hjson";
+			localizationFileWatcher.IncludeSubdirectories = true;
+
+			localizationFileWatcher.Changed += (a, b) => {
+				HandleFileChangedOrRenamed(b.Name);
+			};
+			localizationFileWatcher.Renamed += (a, b) => {
+				HandleFileChangedOrRenamed(b.Name);
+			};
+
+			// Begin watching.
+			localizationFileWatcher.EnableRaisingEvents = true;
+		}
+		catch (Exception) {
+			throw;
+		}
+	}
+
+	private void HandleFileChangedOrRenamed(string fileName)
+	{
+		// fileName is relative to watched folder path
+		string modName = fileName.Split(Path.DirectorySeparatorChar).First();
+		if (ModLoader.ModLoader.TryGetMod(modName, out _)) {
+			watchedMod = modName;
+			watcherCooldown = defaultWatcherCooldown;
+			watchedFile = fileName;
```

Comment:
Probably need some way to support multiple files being saved at the same time. Perhaps a `Set<string>` with a lock?

We could canonicalize the file path here to find out which localization file in a mod we actually need to override.